dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/RouteObjectBundleHook.java

```
@@ -0,0 +1,95 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.dxf2.metadata.objectbundle.hooks;
+
+import static org.hisp.dhis.config.HibernateEncryptionConfig.AES_128_STRING_ENCRYPTOR;
+
+import lombok.AllArgsConstructor;
+
+import org.hisp.dhis.common.auth.ApiTokenAuth;
+import org.hisp.dhis.common.auth.Auth;
+import org.hisp.dhis.common.auth.HttpBasicAuth;
+import org.hisp.dhis.dxf2.metadata.objectbundle.ObjectBundle;
+import org.hisp.dhis.route.Route;
+import org.jasypt.encryption.pbe.PBEStringCleanablePasswordEncryptor;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+/**
+ * @author Morten Olav Hansen
+ */
+@Component
+@AllArgsConstructor
+public class RouteObjectBundleHook
+    extends AbstractObjectBundleHook<Route>
+{
+    @Qualifier( AES_128_STRING_ENCRYPTOR )
+    private final PBEStringCleanablePasswordEncryptor encryptor;
+
+    @Override
+    public void preCreate( Route route, ObjectBundle bundle )
+    {
+        encrypt( route );
+    }
+
+    @Override
+    public void preUpdate( Route route, Route persistedObject, ObjectBundle bundle )
+    {
+        encrypt( route );
+    }
+
+    private void encrypt( Route route )
+    {
+        Auth auth = route.getAuth();
+
+        if ( auth == null )
+        {
+            return;
```

Do we want to silently return null or throw exception?

---

No need to report here, if there is no auth.. just means its a simple non-authorized http request, so no `auth` is fine